example-async.js

```diff
@@ -0,0 +1,158 @@
+'use strict'
+
+const Fastify = require('fastify')
+
+function build (opts) {
+  const fastify = Fastify(opts)
+
+  fastify
+    .register(require('fastify-jwt'), { secret: 'supersecret' })
+    .register(require('fastify-leveldb'), { name: 'authdb-async' })
+    .register(require('./fastify-auth'))
+    .after(routes)
+
+  fastify.decorate('verifyJWTandLevel', verifyJWTandLevel)
+  fastify.decorate('verifyUserAndPassword', verifyUserAndPassword)
+
+  function verifyJWTandLevel (request, reply) {
+    const jwt = this.jwt
+    const level = this.level
+
+    return new Promise(function (resolve, reject) {
```

Comment:
I would not use `new Promise()` here.

Reply:
As for me the idea of the test is to test what if `auth` function returns `Promise`.
How exactly it returns it is out of scope.

I don't mind changing this based on your suggestion though.